packages/ruleset/src/functions/valid-schema-example.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2025 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+const { validate } = require('jsonschema');
+const { validateSubschemas } = require('@ibm-cloud/openapi-ruleset-utilities');
+const { LoggerFactory } = require('../utils');
+
+let ruleId;
+let logger;
+
+module.exports = function (schema, _opts, context) {
+  if (!logger) {
+    ruleId = context.rule.name;
+    logger = LoggerFactory.getInstance().getLogger(ruleId);
+  }
+
+  return validateSubschemas(schema, context.path, checkSchemaExamples);
```

Comment:
@hudlow correct me if I'm wrong but this time, I _do_ think it's appropriate to use `validateSubschemas`, as the goal is to prescribe valid example instances for any simple schema that happens to have an example.